packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -714,6 +714,19 @@ def data_to_lists(data: Dict[str, Any]) -> Dict[str, List[Any]]:
         return {k: [v] for k, v in data.items()}
 
 
+class BaseDBKeys(Enum):
```

Comment:
By introducing an enum we make it more transparent that these are reserved keys in the DB. We can then also apply checks to all the enums across all chained skills to ensure no duplication of keys (which could lead to unintended overwriting).

Reply:
So what about introducing. An extra library? :dancing_women: :fireworks: 

[StrEnum](https://docs.python.org/3/library/enum.html#enum.StrEnum) has been introduced in python 3.11. However, it's already available for python 3.7 onward here https://github.com/irgeek/StrEnum. It is a pure python library without any dependencies of its own. We can use it in combination with `auto()` to ensure the key: value pairs are never out of wack.

Reply:
Hm seems to be offering only marginal utility here. Also keep in mind that this PR is mostly a prep for the actual refactor. It's backward compatible and only intends to make the links more explicit so we have an easier time refactoring.

Reply:
I was hoping it could resolve this issue where mypy always complains that Performative.<SOME_SPECIFIC_PERFORMATIVE> is of type "str" while "Performative" is expected. We have `type: ignore` all throughout our code base for this

Reply:
Good news on this!

One reason for these
https://github.com/valory-xyz/open-autonomy/blob/main/packages/valory/skills/safe_deployment_abci/behaviours.py#L220-L226
is this issue in mypy https://github.com/python/mypy/issues/4717 which can now be circumvented https://github.com/python/mypy/issues/4717#issuecomment-1264388209 . So when we update tomte next we can finally remove the ignores in this place.

I haven't found the exact issue on mypy that causes the failure you describe. But I am 99% sure it is a mypy issue. The types can be manually checked to be correctly declared.